common/osOpen_windows.go

```diff
@@ -6,6 +6,8 @@ import (
 	"os"
 )
 
+// TODO: this is not reliable to use for directories, except inside OSStat (since the stat gets the right result)
+//   (But if the raw result of this routine is used, checking its internal dir flag, that flag is wrong for directories)
 func OSOpenFile(name string, flag int, perm os.FileMode) (*os.File, error) {
```

Comment:
What does it look like, when this incorrectly happens? Is there a known solution?

Reply:
Solution is to use os.Stat(string) instead.  Basically the statue of the os.File is messed up, in that it actually points to directory, but it thinks it points to a File.  (On Windows, Golang uses very different internal representations for files vs directories, inside the implementation of os.File)

Some things just don't work. E.g. the readdir call in the parallel enumerator.  But even for things that do work, I'm not sure I'd want to trust them long term. (Right now, we seem to get away with it, because of how we happen to be using the os.Files, but I don't think we can rely on usage patterns staying the same long term).